FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -49,7 +50,28 @@ public enum ModelDownloadType {
 }
 
 /// Downloader to manage custom model downloads.
-public struct ModelDownloader {
+public class ModelDownloader {
+  /// FirebaseApp associated with this instance of ModelDownloader.
+  private let app: FirebaseApp
+
+  private init(app: FirebaseApp) {
+    self.app = app
+  }
+
+  /// Model downloader with default app.
+  static func modelDownloader() throws -> ModelDownloader {
+    guard let defaultApp = FirebaseApp.app() else {
```

Comment:
Discussed offline: there are challenges consuming the Objective-C API right now and supporting SwiftPM so the Core SDK will need to split to a public and internal API surface. Since that will take some time to design and implement, let's mark a `TODO` here to switch to the component system when ready and implement things manually for now.

Reply:
Done.